docs/source/en/autoregressive_generation.md

```diff
@@ -0,0 +1,243 @@
+
+Autoregressive generation with other modalities behave mostly as described above for LLMs. As such, let's focus on the differences that you may enconter when generating with other modalities:
+* Non-text model inputs rely on the [`AutoProcessor`](https://huggingface.co/docs/transformers/model_doc/auto#transformers.AutoProcessor) class for pre-processing;
+* If the output of your model's forward pass is not a discrete set (e.g. if they are embeddings), then the logit processing step described above does not apply, but there may be custom model output processing steps between iterations.
+
+And... that's it!
+
+For instance, let's take the image captioning problem.
+
+```py
+>>> from PIL import Image
+>>> import requests
+
+>>> url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/coco_sample.png"
+>>> image = Image.open(requests.get(url, stream=True).raw)
+```
+
+The variable `image` contains a lovely image of two cats.
+
+<div class="flex justify-center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/coco_sample.png" alt="Test image"/>
+</div>
+
+You can now use the same workflow as above to caption it, replacing the `AutoTokenizer` by the `AutoProcessor` and importing the appropriate model class.
+
+```py
+>>> from transformers import AutoProcessor, AutoModelForVision2Seq
+
+>>> model = AutoModelForVision2Seq.from_pretrained(
+...     "Salesforce/blip2-opt-2.7b", device_map="auto", load_in_4bit=True
+... )
+>>> processor = AutoProcessor.from_pretrained("Salesforce/blip2-opt-2.7b")
+>>> model_inputs = processor(image, return_tensors="pt").to("cuda")
+
+>>> generated_ids = model.generate(**model_inputs)
+>>> processor.batch_decode(generated_ids, skip_special_tokens=True)[0]
+'two cats laying on a pink couch\n'
+```
+
+
+## Common pitfalls
+
+Autoregressive generation can be controlled with great precision, as we explain in our [generation strategies guide](generation_strategies). However, before you read our advanced docs, let's go through the most common pitfalls, using an LLM as an example.
+
+```py
+>>> from transformers import AutoModelForCausalLM, AutoTokenizer
+
+>>> tokenizer = AutoTokenizer.from_pretrained("openlm-research/open_llama_7b")
+>>> tokenizer.pad_token = tokenizer.eos_token  # Llama has no pad token by default
+>>> model = AutoModelForCausalLM.from_pretrained(
+...     "openlm-research/open_llama_7b", device_map="auto", load_in_4bit=True
+... )
+```
+
+1. Not controlling the maximum length. If not specified in the [`~generation.GenerationConfig`] file, a `generate` call will return up to `20` tokens (our default value). We highly recommend you manually setting `max_new_tokens` in your generate call -- this flag controls the maximum number of new tokens it can return. Please note that LLMs (more precisely, [decoder-only models](https://huggingface.co/learn/nlp-course/chapter1/6?fw=pt)) also return the input prompt as part of the output.
```

Comment:
Subjective opinion - feel free to ignore :) 

Personally, I'd have each of these points as their own sub-header to make it easier to pick our when scrolling the page e.g. 

```suggestion
### 1. Not controlling the maximum length. 
If not specified in the [`~generation.GenerationConfig`] file, a `generate` call will return up to `20` tokens (our default value). We highly recommend you manually setting `max_new_tokens` in your generate call -- this flag controls the maximum number of new tokens it can return. Please note that LLMs (more precisely, [decoder-only models](https://huggingface.co/learn/nlp-course/chapter1/6?fw=pt)) also return the input prompt as part of the output.
```